extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -538,17 +539,17 @@ export class AmpA4A extends AMP.BaseElement {
   /**
    * Try to validate creative is AMP through crypto signature.
    * @param {!ArrayBuffer} creative  Bytes of the entire signed creative.
-   * @param {?ArrayBuffer} signature  Bytes for creative signature (decoded from
+   * @param {?Uint8Array} signature  Bytes for creative signature (decoded from
    *   base64, if necessary.)
-   * @return {!Promise<ArrayBuffer>}  Promise to a guaranteed-valid AMP creative
+   * @return {!Promise<?ArrayBuffer>}  Promise to a guaranteed-valid AMP creative
    *   or null if the creative is unsigned or invalid.
    * @private
    */
   validateAdResponse_(creative, signature) {
     // Validate when we have a signature and we have native crypto.
     if (!signature) {
       // Guaranteed not a AMP creative.
-      return Promise.resolve(null);
+      return /** @type {!Promise<?ArrayBuffer>} */ (Promise.resolve(null));
```

Comment:
As I said above: The compiler cannot do type math with generics. For this reason while

'null' is an acceptable value where 'string|null' is expected, the compiler will not accept `Container<null>` where `Container<string|null>` is expected. I'd ask the compiler team if they intend to fix this.